ctapipe/tools/make_irf.py

```diff
@@ -0,0 +1,385 @@
+        if kind == "gamma":
+            spectrum = spectrum.integrate_cone(0 * u.deg, self.ON_radius * u.deg)
+        events["weight"] = calculate_event_weights(
+            events["true_energy"],
+            target_spectrum=target_spectrum,
+            simulated_spectrum=spectrum,
+        )
+
+        return events
+
+    def get_metadata(self, loader):
+        obs = loader.read_observation_information()
+        sim = loader.read_simulation_configuration()
+
+        # These sims better have the same viewcone!
+        assert sim["max_viewcone_radius"].std() == 0
+        sim_info = SimulatedEventsInfo(
+            n_showers=sum(sim["n_showers"] * sim["shower_reuse"]),
+            energy_min=sim["energy_range_min"].quantity[0],
+            energy_max=sim["energy_range_max"].quantity[0],
+            max_impact=sim["max_scatter_range"].quantity[0],
+            spectral_index=sim["spectral_index"][0],
+            viewcone=sim["max_viewcone_radius"].quantity[0],
+        )
+
+        return (
+            sim_info,
+            PowerLaw.from_simulation(
+                sim_info, obstime=self.obs_time * u.Unit(self.obs_time_unit)
+            ),
+            obs,
+        )
+
+    def load_preselected_events(self):
+        opts = dict(load_dl2=True, load_simulated=True, load_dl1_parameters=False)
+        reduced_events = dict()
+        for kind, file, target_spectrum in [
+            ("gamma", self.gamma_file, PYIRF_SPECTRA[self.gamma_sim_spectrum]),
+            ("proton", self.proton_file, PYIRF_SPECTRA[self.proton_sim_spectrum]),
+            (
+                "electron",
+                self.electron_file,
+                PYIRF_SPECTRA[self.electron_sim_spectrum],
+            ),
+        ]:
+            with TableLoader(file, **opts) as load:
+                Provenance().add_input_file(file)
+                header = self.epp.make_empty_table()
+                sim_info, spectrum, obs_conf = self.get_metadata(load)
+                if kind == "gamma":
+                    self.sim_info = sim_info
+                    self.spectrum = spectrum
+                bits = [header]
+                for start, stop, events in load.read_subarray_events_chunked(
+                    self.chunk_size
+                ):
+                    selected = self.epp.normalise_column_names(events)
+                    selected = selected[self.epp.get_table_mask(selected)]
+                    selected = self.make_derived_columns(
+                        kind, selected, spectrum, target_spectrum, obs_conf
+                    )
+                    bits.append(selected)
+
+                table = vstack(bits, join_type="exact")
+                reduced_events[kind] = table
+
+        select_ON = reduced_events["gamma"]["theta"] <= self.ON_radius * u.deg
+        self.signal = reduced_events["gamma"][select_ON]
+        self.background = vstack([reduced_events["proton"], reduced_events["electron"]])
```

Comment:
To avoid some confusion, I would rename these  to `self.signal_events` and `self.background_events`, since there is also later background that is a histogram

Reply:
renamed